gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -349,4 +584,52 @@ private Map<String, AttributeMetadata> remapAttributeMetadataByResultName(
             .collect(Collectors.toUnmodifiableList()),
         attributeMetadataByIdMap);
   }
+
+  private Optional<org.hypertrace.gateway.service.v1.common.Filter> buildFilter(
+      org.hypertrace.gateway.service.v1.common.Filter filter,
+      AttributeSource source,
+      Map<String, AttributeMetadata> attributeMetadataMap) {
+    if (filter.equals(Filter.getDefaultInstance())) {
+      return Optional.empty();
+    }
+
+    org.hypertrace.gateway.service.v1.common.Operator operator = filter.getOperator();
+    switch (operator) {
+      case UNDEFINED:
+        return Optional.empty();
+      case AND:
+      case OR:
+        return buildCompositeFilter(filter, source, operator, attributeMetadataMap);
+      default:
+        List<AttributeSource> sources =
+            attributeMetadataMap
+                .get(
+                    ExpressionReader.getAttributeIdFromAttributeSelection(filter.getLhs())
+                        .orElseThrow())
+                .getSourcesList();
+        return sources.size() == 1 && sources.contains(source)
```

Comment:
Isn't the function generic? Will it not be used to build QS filters? Where are we building filters need to be pushed down to QS?

Reply:
Should we change the condition to `sources.contains(EDS) ? source.size() == 1 : sources.contain(source)`?

Reply:
Done